src/Lecture4.hs

```diff
@@ -156,8 +177,21 @@ string.
 
 If both strings have the same length, return the first one.
 -}
+compareLength :: [a] -> [a] -> Ordering
+compareLength [] [] = EQ
+compareLength (_ : _) [] = GT
+compareLength [] (_ : _) = LT
+compareLength (_ : xs) (_ : ys) = compareLength xs ys
```

Comment:
Awesome implementation of this function that evaluates up until the shortest list and no further! 🏆